internal/pkg/gateway/api.go

```diff
@@ -71,25 +74,24 @@ func (gs *Server) Evaluate(ctx context.Context, request *gp.EvaluateRequest) (*g
 		gs.logger.Debugw("Sending to peer:", "channel", channel, "chaincode", chaincodeID, "MSPID", endorser.mspid, "endpoint", endorser.address)
 
 		pr, err := endorser.client.ProcessProposal(ctx, signedProposal)
-		if err != nil {
-			logger.Debugw("Evaluate call to endorser failed", "chaincode", chaincodeID, "channel", channel, "txid", request.TransactionId, "endorserAddress", endorser.endpointConfig.address, "endorserMspid", endorser.endpointConfig.mspid, "error", err)
-			errDetails = append(errDetails, errorDetail(endorser.endpointConfig, err))
-			gs.registry.removeEndorser(endorser)
-			endorser = plan.retry(endorser)
+		success, message, retry, remove := gs.responseStatus(pr, err)
+		if success {
+			response = pr.Response
+		} else {
+			logger.Debugw("Evaluate call to endorser failed", "chaincode", chaincodeID, "channel", channel, "txid", request.TransactionId, "endorserAddress", endorser.endpointConfig.address, "endorserMspid", endorser.endpointConfig.mspid, "error", message)
+			errDetails = append(errDetails, errorDetail(endorser.endpointConfig, message))
+			if remove {
+				gs.registry.removeEndorser(endorser)
+			}
+			if retry {
```

Comment:
How many retries? I guess just a single retry?

Reply:
It's in a loop.  It will retry until it's either successful, or runs out of peers to try.